### PR TITLE
Support embedded `go_source` rule type in `go_test` and `go_library` rules

### PR DESCRIPTION
**Pull request by lbcjbb**
# Checklist

- [ ] I have filed an issue about this change and discussed potential changes with the maintainers.
- [ ] I have received the approval from the maintainers to make this change.
- [x] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked.
See the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more details.

# Discussion thread for this change

Issue number: #1265 

# Description of this change

This change mark all source files and generated source files of all embedded `go_source` rules in `go_test` and `go_library` rules as source files.

### CODE REVIEW

**Review by blorente — APPROVED** [review]
Thanks for the contribution!
Would you be open to extending the example project (`examples/go/with_proto`) with a `go_source` before merging this?

**Comment by lbcjbb** [comment]
It's done. Thank you @blorente

File: examples/go/with_go_source/with_go_source.bazelproject [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+additional_languages:
+  go
+
+gazelle_target: //:gazelle
+
+import_run_configurations:
+  # Test that validates that macro expansion works, as per issue:
+  #    https://github.com/bazelbuild/intellij/issues/4112#event-7958662669
+  # This configuration should be executed on both 'run' (the green arrow) and 'debug' (the bug) modes.
+  run_configurations/test_macro_expansion.xml
```
Comment (blorente) [inline examples/go/with_go_source/with_go_source.bazelproject:15]:
```suggestion
```
I don't think we import any run configurations here. This line imports a file from the root of the example project, and it doesn't exist.

**Review by blorente — APPROVED** [review]
Confirmed that it fixes the problem. Thanks for the contribution! Just a small fix on the project view file and we can merge.

File: examples/go/with_go_source/with_go_source.bazelproject (hunk 1 above)
Comment (lbcjbb) [reply examples/go/with_go_source/with_go_source.bazelproject:15]: It's removed. Thanks

**Pull request merged**